src/HealthChecks.Aws.S3/S3HealthCheck.cs

```diff
@@ -1,53 +1,69 @@
-
-        public S3HealthCheck(S3BucketOptions bucketOptions)
-        {
+using Amazon.Runtime;
+using Amazon.S3;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.Aws.S3
+{
+    public class S3HealthCheck : IHealthCheck
+    {
+        private readonly S3BucketOptions _bucketOptions;
+
+        public S3HealthCheck(S3BucketOptions bucketOptions)
+        {
             if (bucketOptions == null)
             {
                 throw new ArgumentNullException(nameof(bucketOptions));
-            }
+            }
+
             if (bucketOptions.S3Config == null)
             {
                 throw new ArgumentNullException(nameof(S3BucketOptions.S3Config));
-            }
-            _bucketOptions = bucketOptions;
-        }
-
-        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
-        {
-            try
-            {
-                bool keysProvided = !string.IsNullOrEmpty(_bucketOptions.AccessKey) &&
-                                    !string.IsNullOrEmpty(_bucketOptions.SecretKey);
-
-                var client = keysProvided
-                    ? new AmazonS3Client(_bucketOptions.AccessKey, _bucketOptions.SecretKey, _bucketOptions.S3Config)
-                    : new AmazonS3Client(_bucketOptions.S3Config);
+            }
+            _bucketOptions = bucketOptions;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                AWSCredentials credentials = _bucketOptions.Credentials;
+                if (credentials == null)
+                {
+#pragma warning disable CS0618 // Type or member is obsolete
+                    if (!string.IsNullOrEmpty(_bucketOptions.AccessKey) && !string.IsNullOrEmpty(_bucketOptions.SecretKey))
+                    {
+                        // for backwards compatibility we create the basic credentials if the old fields are used
+                        // but if they are not specified we fallback to using the default profile
+                        credentials = new BasicAWSCredentials(_bucketOptions.AccessKey, _bucketOptions.SecretKey);
+                    }
+#pragma warning restore CS0618 // Type or member is obsolete
+                }
+
+                bool keysProvided = credentials != null;
+
+#pragma warning disable CS0618 // Type or member is obsolete
+                var client = keysProvided
+                    ? new AmazonS3Client(_bucketOptions.AccessKey, _bucketOptions.SecretKey, _bucketOptions.S3Config)
+                    : new AmazonS3Client(credentials, _bucketOptions.S3Config);
+#pragma warning restore CS0618 // Type or member is obsolete
 
                 using (client)
                 {
                     var response = await client.ListObjectsAsync(_bucketOptions.BucketName, cancellationToken);
-

```

Comment:
I see a bunch of non-functional changes - removing newlines here, adding newlines there.

Reply:
These issues should have been resolved, was a bad merge causing extra newlines in S3HealthCheck.cs

Reply:
I thought these were fixed, but I still see the diff in github. Need to look into what is happening for this file, no idea what it is.

Reply:
So it looks like the issue is with S3HealthCheck.cs in master. It uses a mixture of newlines, which got fixed up when I opened the file.
I could revert this back to the original file, keeping the mixed new lines and then apply my changes if we want to keep this.

Reply:
Understood. OK to use CRLF?

Reply:
Whatever, does not matter.